datafusion/physical-expr/src/aggregate/count_distinct/mod.rs

```diff
@@ -175,9 +172,13 @@ impl PartialEq<dyn Any> for DistinctCount {
     }
 }
 
+/// General purpose distinct accumulator that works for any DataType by using
+/// [`ScalarValue`]. Some types have specialized accumulators that are (much)
+/// more efficient such as [`PrimitiveDistinctCountAccumulator`] and
+/// [`StringDistinctCountAccumulator`]
 #[derive(Debug)]
 struct DistinctCountAccumulator {
-    values: HashSet<DistinctScalarValues, RandomState>,
+    values: HashSet<ScalarValue, RandomState>,
```

Comment:
This was typedefd to
```rust
type DistinctScalarValues = ScalarValue;
```
Which serves no purpose that I can tell other than obscuring what this code is doing